storybook/README.md

```diff
@@ -0,0 +1,322 @@
+
+  /**
+   * Here you can define how the Storybook controls look like. More info
+   * below.
+   */
+  argTypes: Partial< ArgTypes< MyComponentProps > >;
+}
+```
+
+#### Defining controls
+
+As mentioned, Storybook will try to infer the best control for your property type (e.g. an on/off switch for a `boolean`, etc.). However, there are times in which you want different options.
+
+Here is a link to the official Storybook documentation: https://storybook.js.org/docs/react/essentials/controls
+
+But a TL;DR of most common usecases:
+
+-   The shape of the prop looks like so:
+
+```ts
+{
+  [$myPropName]: {
+    control: { type: $controlType },
+    // Only applicable for selects/radios and such
+    options: $controlOptions
+  }
+}
+```
+
+-   You can disable a control like so:
+
+```ts
+{
+  [$myPropName]: { control: false }
+}
+```
+
+#### The Story template
+
+The recommended way to create a story is by creating a template function and duplicating it for each story, to avoid extra scaffolding. The simplest form of the template is as follows:
+
+```tsx
+const Template: Story< MyComponentProps > = ( args ) => (
+	<MyComponent { ...args } />
+);
+```
+
+In this way you are rendering the component in the viewport and binding the component properties to the Storybook controls.
+
+In this template function, however, you might put any extra logic that your component rendering might need. Common use-cases are [wrapping them in context providers](#context-providers), or [simulating controlled components](#controlled-components).
+
+#### Defining Stories
+
+If you have followed this scaffold, defining a new story is as simple as this:
+
+```ts
+export const Default = Template.bind( {} );
+```
+
+Usually, you want a story with the name `Default`, and other stories with interesting variations of your component ([see below](#what-should-constitute-a-story)). In order to create those variations, you should work with some of the properties of this function prototype. Here are the most common:
+
+```ts
+{
+  /**
+   * Define the properties to pass for this specific story. Often,
+   * `Default` stories wouldn't even need this.
+   */
+  args: Partial< MyComponentProps >;
+
+  /**
+   * You will rarely need this, as story names are automatically generated
+   * from your constant name. But that's good to know if you ever need.
+   */
+  storyName: string;
+}
+```
+
+Full official docs: https://storybook.js.org/docs/react/writing-stories/introduction
+
+### Snippets
+
+If you are using [VSCode](https://code.visualstudio.com/), this repo includes some helpful snippets that will become available inside `.ts` and `.tsx` files. You can find them inside `.vscode/storybook.code-snippets`.
+
+The `sbs` (“Storybook story”) will scaffold the entire code in the [section above](#scaffold-tour). If you have respected the naming conventions, it will also property import your component and the properties with the correct names, saving you a bunch of time.
```

Comment:
```suggestion
The `sbs` (“Storybook story”) will scaffold the entire code in the [section above](#scaffold-tour). If you have respected the naming conventions, it will also properly import your component and the properties with the correct names, saving you a bunch of time.
```